### PR TITLE
Chore/GH Actions build matrix

### PR DESCRIPTION
**Pull request by stephengtuggy**
Thank you for submitting a pull request and becoming a contributor to the Vega Strike Core Engine.

Please answer the following:

Code Changes:
- [ ] Have the PR Validation Tests been run? See https://github.com/vegastrike/Vega-Strike-Engine-Source/wiki/Pull-Request-Validation
- [ ] This is a documentation change only
- [x] This is a CI/CD change only

Issues:
- Please list any related issues
#382 

Purpose:
- What is this pull request trying to do? Get our GitHub Actions CI a little closer to what it needs to be
- What release is this for? 0.7.0
- Is there a project or milestone we should apply this to? 0.7.x

### CODE REVIEW

**Comment by stephengtuggy** [comment]
Note that I canceled the Travis CI build manually. It wasn't going to work anyway, due to changes I made to the `cibuild` script.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+            FLAGS:    '-DUSE_PYTHON_3=ON'
+          - FROM:     'fedora:33'
+            COMPILER: 'clang'
+            FLAGS:    '-DUSE_PYTHON_3=ON'
+          - FROM:     'fedora:32'
+            COMPILER: 'clang'
+            FLAGS:    '-DUSE_PYTHON_3=ON'
+          - FROM:     'centos:8'
+            COMPILER: 'clang'
+            FLAGS:    '-DUSE_PYTHON_3=ON'
```
Comment (BenjamenMeyer) [inline .github/workflows/ci.yml:62]: so we lose our Mac builds?

**Review by BenjamenMeyer — APPROVED** [review]
LGTM; but looks like we're losing our Mac builds... not too big an issue since they were not fully functional yet.

Looks like we can get Windows and some Mac builds via GH Actions (https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#jobsjob_idruns-on); but that'd be good for the future (on Master).

**Comment by BenjamenMeyer** [comment]
we'll want to port this and #383 to master as well.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (stephengtuggy) [reply .github/workflows/ci.yml:62]: Yes, we do lose our Mac builds. For now.

**Comment by stephengtuggy** [comment]
> we'll want to port this and #383 to master as well.

Definitely.

On master, perhaps I can add the macOS builds back in. Maybe Windows, too.

**Pull request merged**